nemoguardrails/library/prompt_security/actions.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Prompt/Response protection using Prompt Security."""
+
+import logging
+import os
+from typing import Optional
+
+import httpx
+
+from nemoguardrails.actions import action
+
+log = logging.getLogger(__name__)
+
+
+async def ps_protect_api_async(
+    ps_protect_url: str,
+    ps_app_id: str,
+    prompt: Optional[str] = None,
+    system_prompt: Optional[str] = None,
+    response: Optional[str] = None,
+    user: Optional[str] = None,
+):
+    """Calls Prompt Security Protect API asynchronously."""
```

Comment:
Add a more comprehensive docstring, specially for `ps_protect_url` and `ps_app_id` is desired here.